### PR TITLE
add riscv64 architecture

### PR DESCRIPTION
**Pull request by kroening**
- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 577c24b3590ab3b9f651b54b1df56b89867d116d).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90164694
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

File: src/util/config.cpp [hunk 1]
```diff
@@ -1280,38 +1308,40 @@ irep_idt configt::this_architecture()
...
-    #ifdef __arch64__
-    this_arch="sparc64";
-    #else
-    this_arch="sparc";
-    #endif
-  #elif defined(__ia64__)
+#elif defined(__riscv)
+  this_arch = "riscv64";
+#elif defined(__sparc__)
+#ifdef __arch64__
```
Comment (smowton) [inline src/util/config.cpp]: Suggest to keep the useful indenting of these lines?

Comment (kroening) [reply src/util/config.cpp]: That would require disabling clang-format.

Comment (smowton) [reply src/util/config.cpp]: go on, type `// clang-format off`. Live dangerously!

File: src/util/config.cpp [hunk 2]
```diff
@@ -1281,41 +1310,44 @@ irep_idt configt::this_architecture()
...
     #ifdef __arch64__
-    this_arch="sparc64";
+      this_arch = "sparc64";
     #else
-    this_arch="sparc";
+      this_arch = "sparc";
     #endif
   #elif defined(__ia64__)
-  this_arch="ia64";
+    this_arch="ia64";
```
Comment (smowton) [inline src/util/config.cpp]: Inconsistenct `x=y` vs. `x = y`, otherwise lgtm

**Review by smowton — APPROVED** [review]

**Comment by kroening** [comment]
Ok, now with extra whitespace!

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 73a8447fc8580ba096970485a2a1e58b02baeb11).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90405485
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 58326f3c299e3c511c71db8d1646020598a8fbe3).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90408252

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: fe01b8730b03a6e4d06194ceb504a06c33682915).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90430351

**Pull request merged**